File: leetcode/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int maxArea = 0, ht = 0;
        while(left < right){
            ht = Math.min(height[left], height[right]);
            maxArea = Math.max(maxArea, ht * (right - left));
            if(height[right] > height[left]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxArea;
    }
}

File: leetcode/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        l = 0
        r = n - 1
        max_area = min(height[l], height[r]) * (r - l)

        while l < r:
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
            area = min(height[l], height[r]) * (r - l)
            if area > max_area:
                max_area = area

        return max_area
